ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableSnapshotRefSpec.java

```diff
@@ -89,9 +89,32 @@ public CreateSnapshotRefSpec(String refName, Long snapShotId, Long asOfTime, Lon
     }
 
     public String toString() {
-      return MoreObjects.toStringHelper(this).add("branchName", refName).add("snapshotId", snapshotId)
+      return MoreObjects.toStringHelper(this).add("refName", refName).add("snapshotId", snapshotId)
           .add("asOfTime", asOfTime).add("maxRefAgeMs", maxRefAgeMs).add("minSnapshotsToKeep", minSnapshotsToKeep)
           .add("maxSnapshotAgeMs", maxSnapshotAgeMs).omitNullValues().toString();
     }
   }
+  public static class DropSnapshotRefSpec {
+
+    private String refName;
+    private boolean ifExists;
+
+    public String getRefName() {
+      return refName;
+    }
+
+    public boolean getIfExists() {
+      return ifExists;
+    }
+
+    public DropSnapshotRefSpec(String refName, Boolean ifExists) {
+      this.refName = refName;
+      this.ifExists = ifExists;
+    }
+
+    public String toString() {
+      return MoreObjects.toStringHelper(this).add("refName", refName).add("ifExists", ifExists).omitNullValues()
```

Comment:
```omitNullValues``` in which case this will kick in?

Reply:
Removed the `omitNullValues `. I just want to omit `null `when `explain alter table drop snapshotref`, but here the `refName `& `ifExists` will nerver be null, so `omitNullValues ` is useless.